### ISSUE TITLE
configuring mysql for iam_authentication seems to have a wrong flag name.

### ISSUE THREAD

**Issue by vaikas** [hashicorp/terraform-provider-google#10683]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to the `modular-magician` user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to `hashibot`, a community member has claimed the issue already.

### Terraform Version

Terraform v1.0.11
on darwin_arm64
+ provider registry.terraform.io/hashicorp/google v4.2.0

### Affected Resource(s)

* google_sql_database_instance
The example showing the database_flags configuration for using iam_authentication appears to be wrong. When using the example as shown:
```
    database_flags {
      name  = "cloudsql.iam_authentication"
      value = "on"
    }
```

In particular the example here:
https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/sql_user

Fails with:
```
│ Error: Error, failed to update instance settings for : googleapi: Error 404: The requested flag is either misspelled or unsupported by Cloud SQL., invalidFlagName
```

And looking at the documentation, the correct flag name appears to be  `cloudsql_iam_authentication`
https://cloud.google.com/sql/docs/mysql/create-edit-iam-instances

It was a bit unfortunate that the sql wouldn't return a better error message so had to do bit of a binary search to figure out what flag it was complaining about :)

### Terraform Configuration Files

```tf
# Copy-paste your Terraform configurations here.
#
# For large Terraform configs, please use a service like Dropbox and share a link to the ZIP file.
# For security, you can also encrypt the files using our GPG public key:
#    https://www.hashicorp.com/security
#
# If reproducing the bug involves modifying the config file (e.g., apply a config,
# change a value, apply the config again, see the bug), then please include both:
# * the version of the config before the change, and
# * the version of the config after the change.
```

### Expected Behavior

iam_authentication to be enabled from the example.

### Actual Behavior

Fails with the error message.

### Steps to Reproduce

1. `terraform apply`

### References

* #0000

**Comment by shuyama1** [comment]
Hi, @vaikas! Could you please verify if you are trying to config a MySQL instance or a PostgreSQL instance? Looks like for MySql, `cloudsql_iam_authentication` is the correct flag name, while for PostgreSQL, `cloudsql.iam_authentication` should be used, based on the documentation ([MySQL](https://cloud.google.com/sql/docs/mysql/create-edit-iam-instances#configure-iam-db-instance), [PostgreSQL](https://cloud.google.com/sql/docs/postgres/create-edit-iam-instances#configure-iam-db-instance)).

**Comment by vaikas** [comment]
Yes that was for mysql

**Comment by vaikas** [comment]
Sorry for the terse reply, was on my phone :)

So, yes, this was for mysql indeed, and I'd be happy to send the entire configuration blob that I was using if that helps, but I think it sounds like unfortunately there's just a mismatch in the flag names between different DB implementations? 🤔 

I just went back and looked at the documentation, and there's this item:
```
Example creating a Cloud IAM User. (For MySQL, specify cloudsql_iam_authentication)
```

I'm not sure if this was added later, or I just missed it the first time around. In any case, thanks very much, and looks like the page now states the difference. I'll close this :)
